packages/components/pie-chip/src/index.ts

```diff
@@ -97,31 +110,33 @@ export class PieChip extends LitElement implements ChipProps {
             isLoading,
             isDismissible,
         } = this;
+        const showCloseButton = isDismissible && isSelected;
 
         const classes = {
             'c-chip': true,
             [`c-chip--${variant}`]: true,
             'c-chip--selected': isSelected,
-            'c-chip--dismissible': isDismissible,
+            'c-chip--dismissible': showCloseButton,
+            'c-chip--disabled': disabled,
             'is-loading': isLoading,
         };
 
         return html`
             <div
+                role="${ifDefined(showCloseButton ? undefined : 'button')}"
+                tabindex="${ifDefined(showCloseButton ? undefined : '0')}"
                 aria-atomic="true"
                 aria-busy="${isLoading}"
                 aria-current="${isSelected}"
                 aria-label="${ifDefined(this.aria?.label)}"
                 aria-live="polite"
                 class=${classMap(classes)}
                 data-test-id="pie-chip"
-                tabindex="0"
-                role="button"
-                ?disabled="${disabled}">
+                @click="${this.onClickHandler}">
                     <slot name="icon"></slot>
                     ${isLoading ? this.renderSpinner() : nothing}
                     <slot></slot>
-                    ${isDismissible && isSelected ? this.renderCloseButton() : nothing}
+                    ${showCloseButton ? this.renderCloseButton() : nothing}
```

Comment:
I know its not part of your changes but I wonder if it would be cleaner for the `renderSpinner` and `renderCloseButton` methods to handle returning `nothing` rather than adding conditionals into this main template.

Reply:
personally, I prefer placing the condition outside the handler so it's easier to see what is exactly rendered. Otherwise, I have to look inside the handler to figure out that the close icon ,for example, is conditional and by default not displayed 👀

Reply:
sound good to me 👍 cheers!